### PR TITLE
7.0.0

### PR DESCRIPTION
**Pull request by cottsay**
This change should be **fast-forward** merged after approval.

### CODE REVIEW

File: rclcpp/CHANGELOG.rst [hunk 1]
```diff
@@ -2,6 +2,29 @@
...
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+6.4.0 (2021-03-18)
+------------------
+* Document design decisions that were made for statically typed parameters (`#1568 <https://github.com/ros2/rclcpp/issues/1568>`_)
+* Fix doc typo in CallbackGroup constructor (`#1582 <https://github.com/ros2/rclcpp/issues/1582>`_)
+* Enable qos parameter overrides for the /parameter_events topic  (`#1532 <https://github.com/ros2/rclcpp/issues/1532>`_)
+* Add support for rmw_connextdds (`#1574 <https://github.com/ros2/rclcpp/issues/1574>`_)
+* Remove 'struct' from the rcl_time_jump_t. (`#1577 <https://github.com/ros2/rclcpp/issues/1577>`_)
+* Re-apply `#829 <https://github.com/ros2/rclcpp/issues/829>`_: Add ParameterEventsSubscriber class (`#1573 <https://github.com/ros2/rclcpp/issues/1573>`_)
```
Comment (ivanpauno) [inline rclcpp/CHANGELOG.rst:12]: I typically merge an "add, revert, reapply" sequence in one item, the fact that was temporarily reverted doesn't seem to be worth noting in the changelogs.

File: rclcpp/CHANGELOG.rst [hunk 2]
```diff
@@ -2,6 +2,29 @@
...
+* Add tests for declaring statically typed parameters when undeclared parameters are allowed (`#1575 <https://github.com/ros2/rclcpp/issues/1575>`_)
+* Quiet clang memory leak warning on "DoNotOptimize". (`#1571 <https://github.com/ros2/rclcpp/issues/1571>`_)
+* Revert "Add ParameterEventsSubscriber class (`#829 <https://github.com/ros2/rclcpp/issues/829>`_)" (`#1572 <https://github.com/ros2/rclcpp/issues/1572>`_)
+* Add ParameterEventsSubscriber class (`#829 <https://github.com/ros2/rclcpp/issues/829>`_)
+* When a parameter change is rejected, the parameters map shouldn't be updated.
+* Fix when to throw the NoParameterOverrideProvided exception.
```
Comment (ivanpauno) [inline rclcpp/CHANGELOG.rst:18]: you can add the PR link to these two items, which is https://github.com/ros2/rclcpp/pull/1567

**Review by ivanpauno — APPROVED** [review]

**Comment by wjwwood** [comment]
Comparison: https://github.com/ros2/rclcpp/compare/6.3.1...master

**Comment by wjwwood** [comment]
Should this be a major bump? I don't know if the changes to either the declare parameters functions or the changes to the `struct` usage in the public API constitute an update to the major version according to semver @ivanpauno @mabelzhang

**Comment by ivanpauno** [comment]
> I don't know if the changes to either the declare parameter

I think that a major bump is reasonable considering that  previous code can break, e.g.:

```cpp
node->declare_parameter("asd", 3);
node->set_parameter({"asd", "string"});
```

that was previously valid, now it isn't.

**Review by wjwwood — APPROVED** [review]

**Comment by wjwwood** [comment]
I think we should fast-forward this change to avoid a merge commit. I don't know if we've done that via GitHub in the past or manually.

**Comment by cottsay** [comment]
> I think we should fast-forward this change to avoid a merge commit. I don't know if we've done that via GitHub in the past or manually.

Yeah, I called that out in the initial description. I don't think the web UI gives you any buttons to do that. I've been doing it manually for other repos, and pushing the tag along with it at the same time. Seems to work pretty well.

**Pull request merged**